x-pack/solutions/security/plugins/security_solution/server/lists_integration/endpoint/handlers/exceptions_pre_single_list_find_handler.ts

```diff
@@ -28,7 +28,9 @@ export const getExceptionsPreSingleListFindHandler = (
 
     // Validate Trusted applications
     if (TrustedAppValidator.isTrustedApp({ listId })) {
-      await new TrustedAppValidator(endpointAppContextService, request).validatePreSingleListFind();
+      await new TrustedAppValidator(endpointAppContextService, request).validatePreSingleListFind(
+        data
+      );
       return data;
```

Comment:
i see that so far all the modifications for space awareness have been some kind of *validatons* towards requests, but modifying the filter sent by the client feels like a different responsibility. i mean, i would assume, that a validator does not have any side effect at all.

i'd suggest to separate the *per-space filtering* from *validation*: it could be extracted into its own class, or just function. what do you think about this?

Reply:
Yeah, I thought about this as well and don't really know which way to go. Extracting it out then having to call it will make the code possibly a bit more messy as the ability to reuse the internal methods of the existing class would also have to possibly be extracted or duplicated.  It would also add additional calls in the extension handlers as they would now have to call two different class methods to achieve the desired flow where currently we call one for each type of artifact where all logic is contained.

let me think a bit more about this and get back to you. Going to look over the code to see if it can be broken out cleanly.

Reply:
Looked it over and I will break it out into fucntion utilities and call them from the handlers instead of the validators. I'll push the changes up soon.

Reply:
Ok. did the refactoring you suggested. Please take another look

Reply:
awesome! and even more awesome that you extracted `setOwnerSpaceId()` as well 👏